spring-batch-infrastructure/src/main/java/org/springframework/batch/item/validator/BeanValidatingItemProcessor.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.batch.item.validator;
+
+import javax.validation.Validator;
+
+import org.apache.shiro.util.Assert;
+
+import org.springframework.validation.beanvalidation.LocalValidatorFactoryBean;
+import org.springframework.validation.beanvalidation.SpringValidatorAdapter;
+
+/**
+ * A {@link ValidatingItemProcessor} that uses the Bean Validation API (JSR-303)
+ * to validate items.
+ *
+ * @param <T> type of items to validate
+ * @author Mahmoud Ben Hassine
+ * @since 4.1
+ */
+public class BeanValidatingItemProcessor<T> extends ValidatingItemProcessor<T> {
+
+	/**
+	 * Create a new instance of {@link BeanValidatingItemProcessor} with the
+	 * default configuration.
+	 * @throws Exception If unable to configure the Bean Validation infrastructure
+	 */
+	public BeanValidatingItemProcessor() throws Exception {
+		this(new LocalValidatorFactoryBean());
+	}
+
+	/**
+	 * Create a new instance of {@link BeanValidatingItemProcessor}.
+	 * @param localValidatorFactoryBean used to configure the Bean Validation validator
+	 * @throws Exception If unable to configure the Bean Validation infrastructure
+	 */
+	public BeanValidatingItemProcessor(LocalValidatorFactoryBean localValidatorFactoryBean) throws Exception {
+		super();
+		Assert.notNull(localValidatorFactoryBean, "localValidatorFactoryBean must not be null");
+		localValidatorFactoryBean.afterPropertiesSet();
+		Validator validator = localValidatorFactoryBean.getValidator();
+		SpringValidatorAdapter springValidatorAdapter = new SpringValidatorAdapter(validator);
+		SpringValidator<T> springValidator = new SpringValidator<>();
+		springValidator.setValidator(springValidatorAdapter);
+		springValidator.afterPropertiesSet();
```

Comment:
Since the `ValidatingItemProcessor` implements `IntializingBean`, this `afterPropertiesSet` should be called in an override of the parent's method to honor the lifecycle.

Reply:
fixed in d2a00f5